src/bin/metadata.rs

```diff
@@ -0,0 +1,54 @@
+extern crate cargo;
+extern crate docopt;
+extern crate rustc_serialize;
+extern crate toml;
+
+
+use cargo::ops::{output_metadata, OutputTo, OutputFormat, OutputMetadataOptions};
+use cargo::util::{CliResult, CliError, Config};
+use cargo::util::important_paths::find_root_manifest_for_wd;
+
+#[derive(RustcDecodable)]
+struct Options {
+    flag_features: Vec<String>,
+    flag_manifest_path: Option<String>,
+    flag_no_default_features: bool,
+    flag_output_format: OutputFormat,
+    flag_output_path: OutputTo,
+    flag_verbose: bool,
+}
+
+pub const USAGE: &'static str = "
+Output the resolved dependencies of a project, the concrete used versions
+including overrides, in machine-readable format.
+
+Usage:
+    cargo metadata [options]
+
+Options:
+    -h, --help               Print this message
+    -o, --output-path PATH   Path the output is written to, otherwise stdout is used
+    -f, --output-format FMT  Output format [default: toml]
+                             Valid values: toml, json
+    --features FEATURES      Space-separated list of features
+    --no-default-features    Do not include the `default` feature
+    --manifest-path PATH     Path to the manifest
```

Comment:
This may also want to support a `--target` argument as that'll affect the structure of the dependency graph. This should also support the standard `-q` and `--color` flag we have now